src/transformStateWithClones.js

```diff
@@ -7,7 +7,41 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateClones = [];
+
+  for (const action of actions) {
+    const currentState = {};
```

Comment:
The currentState object is being recreated for every action. A more efficient way would be to clone the state object once before the for loop and then modify that clone based on the actions.